packages/core/crates/core-types/src/announcement.rs

```diff
@@ -129,89 +155,66 @@ impl Display for AnnouncementData {
 mod tests {
     use crate::announcement::{AnnouncementData, KeyBinding};
     use core_crypto::keypairs::{Keypair, OffchainKeypair};
+    use hex_literal::hex;
     use multiaddr::Multiaddr;
-    use utils_types::primitives::Address;
-    use utils_types::traits::PeerIdLike;
+    use utils_types::{
+        primitives::Address,
+        traits::{BinarySerializable, PeerIdLike},
+    };
+
+    lazy_static::lazy_static! {
+        static ref KEY_PAIR: OffchainKeypair = OffchainKeypair::from_secret(&hex!("60741b83b99e36aa0c1331578156e16b8e21166d01834abb6c64b103f885734d")).unwrap();
+        static ref CHAIN_ADDR: Address = Address::from_bytes(&hex!("78392d47e3522219e2802e7d6c45ee84b5d5c185")).unwrap();
+        static ref SECOND_KEY_PAIR: OffchainKeypair = OffchainKeypair::from_secret(&hex!("c24bd833704dd2abdae3933fcc9962c2ac404f84132224c474147382d4db2299")).unwrap();
+    }
 
     #[test]
     fn test_key_binding() {
-        let kp = OffchainKeypair::random();
-
-        let kb_1 = KeyBinding::new(Address::default(), &kp);
+        let kb_1 = KeyBinding::new(*CHAIN_ADDR, &KEY_PAIR);
         let kb_2 = KeyBinding::from_parts(kb_1.chain_key.clone(), kb_1.packet_key.clone(), kb_1.signature.clone())
             .expect("should verify correctly");
 
         assert_eq!(kb_1, kb_2, "must be equal");
     }
 
     #[test]
-    fn test_announcement_data() {
-        let keypair = OffchainKeypair::random();
-        let key_binding = KeyBinding::new(Address::default(), &keypair);
-        let peer_id = keypair.public().to_peerid();
-
+    fn test_announcement() {
```

Comment:
Can we please add tests with empty and malformed multiaddrs? Also `/dns/...` MA test to be future proof.